### PR TITLE
improve: confirm password mismatch returns to first password prompt

### PR DESCRIPTION
**Pull request by kimskimchi**
## Summary

If password confirm does not match the first one, right now it prompts only the second one. 
This change remprompts both: 

**Old:**
```
  Password: abc
  Confirm password: abcd
Passwords do not match. 
  Confirm password: abcd
```

**Now:**
```
  Password: abc
  Confirm password: abcd
Passwords do not match. Please try again.
  Password: ...
  Confirm password: ...
```

**Reason**
During admin signup, if user mistypes first password, they have no clear way to fix it. 

**Additional small changes**
- merges `promptUpdatePassword` and `promptConfirmPassword` because it only included a message `fmt.println`. Moved that msg out to the respective functions that call it, since the message is different each time.

### CODE REVIEW

File: internal/cmd/identities.go [hunk 1]
```diff
@@ -289,14 +280,19 @@ func promptPasswordConfirm(oldPassword string) (string, error) {
...
-			Validate: checkConfirmPassword(&passwordConfirm.Password),
+			Validate: checkConfirmPassword(),
 		},
 	}
 
 	if err := survey.Ask(prompts, &passwordConfirm, survey.WithStdio(os.Stdin, os.Stderr, os.Stderr)); err != nil {
 		return "", err
 	}
 
+	if passwordConfirm.Password != passwordConfirm.Confirm {
```
Comment (kimskimchi) [inline internal/cmd/identities.go:291]: @ssoroka moved to here

File: internal/cmd/identities.go [hunk 2]
```diff
@@ -289,14 +280,19 @@ func promptPasswordConfirm(oldPassword string) (string, error) {
 		{
 			Name:     "Confirm",
 			Prompt:   &survey.Password{Message: "Confirm Password:"},
-			Validate: checkConfirmPassword(&passwordConfirm.Password),
+			Validate: checkConfirmPassword(),
```
Comment (BruceMacD) [inline internal/cmd/identities.go:283]:
```suggestion
```

Looks like this validation should be removed from the survey input completely if it happens after the input

File: internal/cmd/identities.go [hunk 3]
```diff
@@ -319,17 +315,12 @@ func checkPasswordRequirements(oldPassword string) survey.Validator {
 	}
 }
 
-func checkConfirmPassword(password *string) survey.Validator {
+func checkConfirmPassword() survey.Validator {
```
Comment (BruceMacD) [inline internal/cmd/identities.go:318]: I think this function should be removed entirely

**Review by BruceMacD — COMMENTED** [review]
Good catch on this bad behavior, I think the validator function to confirm that the passwords match here that can be cleaned up. In the case that the validator function test could be removed too.

File: internal/cmd/identities.go (hunk 3 above)
Comment (kimskimchi) [reply internal/cmd/identities.go:318]: remove the type check?

Comment (BruceMacD) [reply internal/cmd/identities.go:318]: Yeah, anything the survey library lets them enter should be a valid string (it doesn't allow nil response)

File: internal/cmd/identities.go (hunk 2 above)
Comment (kimskimchi) [reply internal/cmd/identities.go]: done

**Review by BruceMacD — APPROVED** [review]

**Pull request merged**